linker/shared/src/test/scala/org/scalajs/linker/frontend/optimizer/IntegerDivisionsTest.scala

```diff
@@ -0,0 +1,639 @@
+      assertEquals(
+          divisor.toString(),
+          MagicData(expectedM, expectedAdd, expectedShift),
+          computeUnsignedMagic(divisor))
+    }
+
+    // Test cases from Hacker's Delight, 10-14, Table 10-1, Unsigned column
+
+    test(0xaaaaaaab, 0, 1, 3)
+    test(0xcccccccd, 0, 2, 5)
+    test(0xaaaaaaab, 0, 2, 6)
+    test(0x24924925, 1, 3, 7)
+    test(0x38e38e39, 0, 1, 9)
+    test(0xcccccccd, 0, 3, 10)
+    test(0xba2e8ba3, 0, 3, 11)
+    test(0xaaaaaaab, 0, 3, 12)
+    test(0x51eb851f, 0, 3, 25)
+    test(0x10624dd3, 0, 3, 125)
+    test(0xd1b71759, 0, 9, 625)
+
+    // For completeness, but we never use the results of this function for powers of 2
+    for (k <- 1 to 31)
+      test(1 << (32 - k), 0, 0, 1 << k)
+  }
+
+  @Test def testComputeSignedMagicLong(): Unit = {
+    def test(expectedM: Long, expectedAdd: Int, expectedShift: Int, divisor: Long): Unit = {
+      assertEquals(
+          divisor.toString(),
+          MagicData(expectedM, expectedAdd, expectedShift),
+          computeSignedMagic(Math.abs(divisor), divisor < 0L))
+    }
+
+    // Test cases from Hacker's Delight, 10-14, Table 10-2, Signed column
+
+    test(0x9999999999999999L, 0, 1, -5L)
+    test(0x5555555555555555L, -1, 1, -3L)
+    test(0x5555555555555556L, 0, 0, 3L)
+    test(0x6666666666666667L, 0, 1, 5L)
+    test(0x2aaaaaaaaaaaaaabL, 0, 0, 6L)
+    test(0x4924924924924925L, 0, 1, 7L)
+    test(0x1c71c71c71c71c72L, 0, 0, 9L)
+    test(0x6666666666666667L, 0, 2, 10L)
+    test(0x2e8ba2e8ba2e8ba3L, 0, 1, 11L)
+    test(0x2aaaaaaaaaaaaaabL, 0, 1, 12L)
+    test(0xa3d70a3d70a3d70bL, 1, 4, 25L)
+    test(0x20c49ba5e353f7cfL, 0, 4, 125L)
+    test(0x346dc5d63886594bL, 0, 7, 625L)
+
+    // For completeness, but we never use the results of this function for powers of 2
+    for (k <- 1 to 62) {
+      test(0x7fffffffffffffffL, -1, k - 1, -(1L << k))
+      test(0x8000000000000001L, 1, k - 1, 1L << k)
+    }
+  }
+
+  @Test def testComputeUnsignedMagicLong(): Unit = {
+    def test(expectedM: Long, expectedAdd: Int, expectedShift: Int, divisor: Long): Unit = {
+      assertEquals(
+          divisor.toString(),
+          MagicData(expectedM, expectedAdd, expectedShift),
+          computeUnsignedMagic(divisor))
+    }
+
+    // Test cases from Hacker's Delight, 10-14, Table 10-2, Unsigned column
+
+    test(0xaaaaaaaaaaaaaaabL, 0, 1, 3L)
+    test(0xcccccccccccccccdL, 0, 2, 5L)
+    test(0xaaaaaaaaaaaaaaabL, 0, 2, 6L)
+    test(0x2492492492492493L, 1, 3, 7L)
+    test(0xe38e38e38e38e38fL, 0, 3, 9L)
+    test(0xcccccccccccccccdL, 0, 3, 10L)
+    test(0x2e8ba2e8ba2e8ba3L, 0, 1, 11L)
+    test(0xaaaaaaaaaaaaaaabL, 0, 3, 12L)
+    test(0x47ae147ae147ae15L, 1, 5, 25L)
+    test(0x0624dd2f1a9fbe77L, 1, 7, 125L)
+    test(0x346dc5d63886594bL, 0, 7, 625L)
+
+    // For completeness, but we never use the results of this function for powers of 2
+    for (k <- 1 to 63)
+      test(1L << (64 - k), 0, 0, 1L << k)
+  }
+
```

Comment:
Note to self: continue here.

Reply:
The split destroyed this information. It was located below the magic tests, but above the shape tests and fuzz tests.